### PR TITLE
Adding linting check for core-build-task

### PR DESCRIPTION
**Pull request by frgarc**
### Description

- Adding basic linting check for core-build-task. 
  - Adding triple equality rule.
- Enabling prettier check for math library.
- Enabling ESLint check at Visual Studio Code (for flat config).

### CODE REVIEW

File: libraries/math/package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "@minecraft/core-build-tasks": "*",
     "@minecraft/tsconfig": "*",
     "just-scripts": "^2.2.1",
+    "prettier": "^2.8.2",
```
Comment (SBLMikeDemone) [inline libraries/math/package.json:39]: Just to make sure I understand this correctly, to make it so the lint task uses prettier you only need to add prettier as a dev dependency? If prettier isn't a dev dependency does the lint task normally fail?

Comment (frgarc) [reply libraries/math/package.json:39]: It does not fail, but it is skipped:
image

Comment (SBLMikeDemone) [reply libraries/math/package.json:39]: huh, cool!

File: prettier.config.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
     arrowParens: 'avoid',
     bracketSpacing: true,
-    endOfLine: 'crlf',
```
Comment (SBLMikeDemone) [inline prettier.config.js:4]: line ending changes scare me after hitting issues with them in the docs generator repo, why was this necessary?

Comment (frgarc) [reply prettier.config.js:4]: It is same problem we had with the samples repo, depending OS the end of line is different. In this case PR was not passing the lint checks as end of line was detected different (it runs on Linux).
Found that flipping this setting avoids this problem.

Comment (SBLMikeDemone) [reply prettier.config.js:4]: Ah, alright. To align these settings with the docs generator and minecraft, you could keep this as crlf and add a `.gitattributes` file with `* text eol=crlf`. Then prettier should behave the same as our other ones.

Comment (frgarc) [reply prettier.config.js]: Ok, changing to this mechanism

File: tools/core-build-tasks/package.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
   "scripts": {
     "build-tools": "just-scripts build-tools",
     "clean-tools": "just-scripts clean-tools",
+    "lint": "just lint",
```
Comment (rlandav) [inline tools/core-build-tasks/package.json:22]: nit: `just-scripts lint`

Just a tad bit more portable for some platforms or install set ups

**Review by rlandav — APPROVED** [review]

**Comment by rlandav** [comment]
Let me know when you want me to merge!

File: tools/core-build-tasks/package.json (hunk 3 above)
Comment (frgarc) [reply tools/core-build-tasks/package.json]: Fixed, thanks
